src/main/webapp/feed.js

```diff
@@ -282,18 +283,19 @@ function fillBadge(currentStep, totalSteps) {
 }
 
 //start here to implement (note for me)
-function setChallengesNavBar(challenges) {
+function setChallengesNavBar() {
     const navBar = document.getElementById("challenges-nav-bar");
     navBar.innerHTML = "<p id='challenges-nav-bar-header'>Challenges</p>";
 
-    for (challenge of challenges) {
-        navBar.appendChild(createChallengeNavBarItem(challenge));
+    //challenge [key, value]
+    for (challenge of challengeMap) {
+        navBar.appendChild(createChallengeNavBarItem(challenge[1]));
 
         const itemBackground = document.createElement('div');
-        itemBackground.id = "challenges-nav-bar-item-background-" + challenge.get("id");
+        itemBackground.id = "challenges-nav-bar-item-background-" + challenge[0];
         itemBackground.className = "challenges-nav-bar-item-background";
         let percentDone = 0;
-        if (challenge.get("steps").length != 0) percentDone = user.challenge_statuses[challenge.get("id")] / challenge.get("steps").length;
+        if (challenge[1].steps.length != 0) percentDone = user.challenge_statuses[challenge[0]] / challenge[1].steps.length;
```

Comment:
What do challenge[0] / challenge[1] refer to?

Reply:
to iterate over my map keys in my for of loop,  js returns array [key, value]. Therefore, to access the key, we write challenge[0] and likewise for the value, challenge[1] . The value represents the challenge object.